light_control.h

```diff
@@ -0,0 +1,126 @@
+#include <Adafruit_NeoPixel.h>
+#include <time.h>
+
+#define SENSOR_PIN  6
+#define PIN 8   // light control pin
+#define PIN_NUM 2 //count of leds
+
+#define WAIT_FOR_SOMEONE 0
+#define INCREASE_BRIGHTNESS 1
+#define LIGHT_ON_WAIT 2
+#define DECREASE_BRIGHTNESS 3
+
+const int statesCount = 4;
+
+/*
+ current state of system:
+ 0 -- waiting to turn light
+ 1 -- turning on light
+ 2 -- waiting to turn off light
+ 3 -- checking sensor and keep illuminate if something detected
+ */
+int state = WAIT_FOR_SOMEONE;
+
+
+class Light_control{
+private:
+    /*
+     time shifts between iterations in each state in mills
+     in state 0 we just waiting for sensor activation, that's why state_duration[0] = NULL
```

Comment:
`NULL` macro usually used with pointers. In this case `unsigned long` values, so, just `0`.

Reply:
I have just tried to pay attention, that this value is not for usage. But I agree, it's to strange. I'll change `NULL` to `0`